src/Google.Api.CommonProtos/Type/MoneyPartial.cs

```diff
@@ -0,0 +1,23 @@
+﻿/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using System;
+
+namespace Google.Type
+{
+    public partial class Money
+    {
+        /// <summary>

```

Comment:
I wonder whether a `DecimalValue` property would be useful here.

You could then move the `ToMoney` extension method into the setter, and avoid the extension at all. Creating a `Money` from a `decimal` would just be:

```csharp
Money money = new Money { DecimalValue = value, CurrencyCode = currency };
```

That doesn't chain as well, but I think I prefer it over the extension method. Thoughts?

Reply:
Thanks for reviewing this.  I am somewhat partial to the extension method as it means significantly less code for the consumer of this library in each of the call sites and I think it's a bit more discoverable/obvious, but I do like the idea of the `DecimalValue` property.  Would you want to do both?  I will discuss this with @GR3G0R by tomorrow.

If we do not go the property route, a null `currencyCode` will already throw via the auto-generated code (though it will do some arithmetic first that will be wasted).  I prefer to allow `string.Empty` currency codes in anything that is implemented for this as this empty strings are permitted by the proto.

Thank you again!!